holmes/core/tool_calling_llm.py

```diff
@@ -562,6 +565,42 @@ def _invoke_tool(
             if not tool_response:
                 tool_response = tool.invoke(tool_params, tool_number=tool_number)
 
+            # Handle approval required status
+            if tool_response.status == ToolResultStatus.APPROVAL_REQUIRED:
+                if self.approval_callback is not None:
+                    command = tool_response.invocation or str(tool_params)
+                    error_message = tool_response.error or "Command requires approval"
+
+                    approved, feedback = self.approval_callback(command, error_message)
+
+                    if approved:
```

Comment:
I think that we might have issue here. What will happen if multiple tool request will result in approval request? Each one of them is from different thread and both of them updating the cli. 

Were you able to bump into this kind of situation? if not worth to try and simulate it and see how it beheaves.

Reply:
Good catch. This is now fixed. All tools are executed and then there is a synchronous handling for any tool that requires approval